xhr.js

```diff
@@ -0,0 +1,30 @@
+var APP = APP || {};
+(function(app) {
+
+    var API_URLROOT = 'https://api.stackexchange.com/2.2/questions/';
+
+    app.xhr = {};
+
+    app.xhr.fetch = function(ids) {
+        return new Promise(function(resolve, reject) {
+            var req = new XMLHttpRequest();
+            var url = app.xhr.makeApiUrl(ids);
+            req.onreadystatechange = function() {
+                if (req.readyState === 4) {
+                    if (req.status === 200) {
+                        return resolve(JSON.parse(req.responseText));
```

Comment:
With the correct type, you can use `req.response` and drop the `JSON.parse`. See https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Sending_and_Receiving_Binary_Data